### PR TITLE
seed/seedwriter: fail early when system seed directory exists

### PR DESCRIPTION
**Pull request by bboozzoo**
Fail early when creating a Core 20 seed when the target directory exists. This
helps to prevent a situation when creating seed gets interrupted and gets
restarted.

### CODE REVIEW

**Review by anonymouse64 — APPROVED** [review]
lgtm

File: seed/seedwriter/seed20.go [hunk 1]
```diff
@@ -159,7 +159,10 @@ func (tr *tree20) mkFixedDirs() error {
 		return err
 	}
 
-	return os.MkdirAll(tr.systemDir, 0755)
+	if err := os.MkdirAll(filepath.Dir(tr.systemDir), 0755); err != nil {
+		return err
+	}
+	return os.Mkdir(tr.systemDir, 0755)
```
Comment (pedronis) [inline seed/seedwriter/seed20.go:165]: what will we have to do if hit this error at runtime?  delete the directory tree?  do we know enough from this to distinguish it?

**Review by pedronis — COMMENTED** [review]
a couple of questions  (the one about the tests is not a blocker)

File: seed/seedwriter/seed20.go (hunk 1 above)
Comment (bboozzoo) [reply seed/seedwriter/seed20.go:165]: There's a bit more context needed which is not available at this level. However, when creating a recovery system in devicestate, we can make some choices about further course of action.

Comment (pedronis) [reply seed/seedwriter/seed20.go:165]: I still think we should turn the directory exists error from here into something specific like a ErrSystemAlreadyExists

**Review by pedronis — COMMENTED** [review]
I would prefer a dedicate error

File: seed/seedwriter/seed20.go (hunk 1 above)
Comment (bboozzoo) [reply seed/seedwriter/seed20.go]: This is done now.

File: seed/seedwriter/writer.go [hunk 2]
```diff
@@ -421,12 +421,28 @@ func (w *Writer) SetOptionsSnaps(optSnaps []*OptionsSnap) error {
 	return nil
 }
 
+// ErrSystemAlreadyExists is an error returned when given seed system already
+// exists.
+type ErrSystemAlreadyExists struct {
```
Comment (pedronis) [inline seed/seedwriter/writer.go:426]: error structs need to be called SystemAlreadyExistsError

**Review by pedronis — COMMENTED** [review]
looks good but error doesn't follow naming conventions

**Review by pedronis — APPROVED** [review]
thanks

**Review by anonymouse64 — APPROVED** [review]
still lgtm

**Comment by bboozzoo** [comment]
Feels like restarting the spread jobs for the 3rd time is wasteful. @mvo5 can you merge this PR? The failure on 14.04 is unrelated.

**Pull request merged**